storage_scrubber/src/checks.rs

```diff
@@ -377,11 +376,54 @@ pub(crate) fn parse_layer_object_name(name: &str) -> Result<(LayerName, Generati
     }
 }
 
+/// Note (<https://github.com/neondatabase/neon/issues/8872>):
+/// Since we do not gurantee the order of the listing, we could list layer keys right before
+/// pageserver `RemoteTimelineClient` deletes the layer files and then the index.
+/// In the rare case, this would give back a transient error where the index key is missing.
+///
+/// To avoid generating false positive, we try streaming the listing for a second time.
 pub(crate) async fn list_timeline_blobs(
     remote_client: &GenericRemoteStorage,
     id: TenantShardTimelineId,
     root_target: &RootTarget,
 ) -> anyhow::Result<RemoteTimelineBlobData> {
+    let res = list_timeline_blobs_impl(remote_client, id, root_target).await?;
+    match res {
+        ListTimelineBlobsResult::Ready(data) => Ok(data),
+        ListTimelineBlobsResult::MissingIndexPart(_) => {
```

Comment:
I don't think it's in general a good idea to simply do such meta operation twice. Would be better if we skip the tenants with missing index inside `list_timeline_blobs`? (As in my PR)

Reply:
Yes, but this could become a false positive and trigger the alert? On the other hand, if we choose to not add this as a fatal error, we could also miss some serious problem.

Reply:
well, at a second thought, it makes some sense to retry twice here to handle deleted branches...

Reply:
Sounds good, I will still incorporate your PR: we will add it as a timeline check error if it failed the download the second time.

Reply:
Incoporated #9068: daf22c39ad2c7051e6a51af9fc741eba8a1c03cd